lib/configcat/configfetcher.rb

```diff
@@ -59,13 +75,71 @@ def get_configuration_json()
       etag = response["ETag"]
       @_etag = etag unless etag.nil? || etag.empty?
       ConfigCat.logger.debug "ConfigCat configuration json fetch response code:#{response.code} Cached:#{response['ETag']}"
-      return FetchResponse.new(response)
+      fetch_response = FetchResponse.new(response)
+
+      # If there wasn't a config change, we return the response.
+      if !fetch_response.is_fetched()
+        return fetch_response
+      end
+
+      preferences = fetch_response.json().fetch(PREFERENCES, nil)
+      if preferences === nil
+        return fetch_response
+      end
+
+      base_url = preferences.fetch(BASE_URL, nil)
+
+      # If the base_url is the same as the last called one, just return the response.
+      if base_url.equal?(nil) || @_base_url == base_url
+        return fetch_response
+      end
+
+      redirect = preferences.fetch(REDIRECT, nil)
+      # If the base_url is overridden, and the redirect parameter is not 2 (force),
+      # the SDK should not redirect the calls and it just have to return the response.
+      if @_base_url_overridden && redirect != RedirectMode::FORCE_REDIRECT
+        return fetch_response
+      end
+
+      # The next call should use the base_url provided in the config json
+      @_base_url = base_url
+      _create_http()
```

Comment:
Consider 
- moving this to the beginning of the get_configuration_json()
- removing _create_http() from the constructor
- in the def _create_http() initialize the @_http only if it is nil or the baseurl (@_http host and port parameters) changed.

Reply:
DONE